transformers/suspicious.go

```diff
@@ -65,6 +65,11 @@ func (p *SuspiciousTransform) CheckIfSuspicious(dm *dnsutils.DnsMessage) {
 		return
 	}
 
+	// Only treat replies or we'll get count most categories twice.
+	if dm.DNS.Type == dnsutils.DNS_TYPE_QUERY {
```

Comment:
To discard queries, it willl be better to add the "filtering" transform like that:

```
transforms:
  filtering:
    log-queries: false
```

Reply:
It sounds like it would work but:
- Wouldn't affect all metrics logged in this collector? In this case we want to only filter queries for the suspicious transformer.
- I'm not sure I fundamentally agree that everyone using the suspicious module would need to specify this extra transform to get accurate reporting.

Reply:
If someone only logs queries, the suspicious transformer will not work, why add this limitation to replies when it's also possible to make this analysis on queries ?
In another hand, I agree with you, this extra config is not evident, perhaps more documentations be sufficient ?

Regarding your first point, It should be possible but it's depend of your config, below a basic example:

```
multiplexer:
  collectors:
    - name: tap
      dnstap:
        listen-ip: 0.0.0.0
        listen-port: 6000

  loggers:
    - name: stdout
      console:
        mode: text
    - name: restapi
      restapi:
        listen-ip: 0.0.0.0
        listen-port: 8080
        transforms:
           filtering:
             log-queries : false
           suspicious:
             
  routes:
    - from: [ tap ]
      to: [ stdout, restapi ]
```

In this example, the discard of queries is only applied on the restpi logger.

Reply:
I get what you're saying. I think it's going to be difficult to detect every configuration incompatibility.

Would it actually even work to use the latency transformer along with filtering log-queries? I haven't tested it but it sounds like it wouldn't?

Maybe the solution is to actually split slow-domains out of suspicious into its own feature since it's the only one that strictly requires latency information to be present. This doesn't solve the double accounting for other categories but that can be fixed by looking at queries only (not responses) and documenting that behavior.

Reply:
I make some tests and this PR https://github.com/dmachard/go-dnscollector/pull/305 must be merged for the previous example to work properly.

Reply:
Just to add more details 
with the following config and PR #305, it's works fine
This approach can be used in your case ?

```
multiplexer:
  collectors:
    - name: tap
      dnstap:
        listen-ip: 0.0.0.0
        listen-port: 6000
      transforms:
        normalize:
          qname-lowercase: true
        latency:
          measure-latency: true
          queries-timeout: 2

  loggers:
    - name: console
      stdout:
        mode: text
        text-format: timestamp-rfc3339ns identity operation rcode qname qtype latency suspicious-score
      transforms:
        filtering:
          log-queries : false
        suspicious:
          threshold-qname-len: 4

  routes:
    - from: [ tap ]
      to: [ console ]
```

Reply:
There's two issues that this PR is trying to address:
- Flagging domains for which resolution takes over some amount of time: The more I think about it, the more I think this should live in the existing "latency" module. We can define the threshold in that config and add a LatencyHigh flag that can be leveraged in the loggers (e.g. prometheus).
- Fixing "suspicious" accounting. I'm going to drop that part since you believe it should be fixed in the config and I don't care enough to argue :-)

Reply:
Now I'm noticing that Latency can be set outside of the "latency" module (e.g. "powerdns" module) so I guess it's not that easy.

Reply:
Just remove "Discard QUERY packets to avoid double accounting." to merge it.

Reply:
Done.